server/data/projects.js

```diff
@@ -94,6 +103,65 @@ const createProject = async (projectObjParam, user) => {
 	return createdProject;
 };
 
+const updateProject = async (projectObjParam, id, user) => {
+	id = isValidObjectId(id);
+	let projectCheck = await getProjectById(id);
+	ownercheck = checkuseraccess(user, projectCheck.owner);
+	if (!ownercheck)
+		throw forbiddenErr(
+			`Not Authorised to update this project. Not Project Owner`
+		);
+	const projectObj = isValidProjectObject(projectObjParam);
+	let { name, description, github, media, technologies, deploymentLink } =
+		projectObj;
+		name = isValidProjectName(name);
```

Comment:
i think `isValidProjectObj` will validate all these checks internally, you don't have to them again here